src/afl-fuzz.c

```diff
@@ -2282,6 +2306,7 @@ int main(int argc, char **argv_orig, char **envp) {
       afl->fsrv.map_size = new_map_size;
       afl->fsrv.trace_bits =
           afl_shm_init(&afl->shm, new_map_size, afl->non_instrumented_mode);
+      afl->fsrv.shm_id = afl->shm.shm_id;
```

Comment:
this will not work in cmplog mode then. the fork server there needs the correct shmid

Reply:
also the map size is usually different in a cmplog forkserver target

Reply:
I think we may have to disable custom_fuzz_run in cmplog mode...
Since we have only one custom_mutator instance, it might not designed to run two binaries (the normal one plus the cmplog one).
Other existing custom_mutator functions may have the same problem? Should we create a separate instance for cmplog?

Reply:
cmplog is a highly valuable asset to have when fuzzing. not sure how this could be done for your case, might need you to experiment a bit.
it has to be a second process, even if it is the same binary (you can use a cmplog binary also for normal fuzzing, it is `just` slower).

Reply:
Maybe you want to have a `custom_fuzz_run_cmplog` then?

Reply:
> Maybe you want to have a `custom_fuzz_run_cmplog` then?

This would work, but I still have some question:
Other existing custom_mutator functions, which save some internal state, may have the same problem?  The mutator it might not designed to run two binaries (the normal one plus the cmplog one).
Should we create a separate mutator instance for cmplog?
At least, I think this should be documented as a special note if we leave it as-is.

Do you have any idea? Also cc @vanhauser-thc